Simulator/simulator.py

```diff
@@ -295,489 +389,367 @@ def calc_system_state_features(self, waiting_job, current_reserve_jobs: SortedLi
-                    if time_waited_sec > max_wait_time_sec:
+                    if time_waited_sec > (max_wait_time_sec):
                         # Remove the job from the queue
                         self.wait_queue.popleft()
-                        self.mark_as_on_demand_resource_job(waiting_job, max_wait_time_sec)
+                        total_on_demand_cost += self.calc_on_demand_cost_for_job(waiting_job, waiting_job_runtime)
+                        self.set_waiting_time(waiting_job, max_wait_time_sec)
+                        waited_max_time += 1
                     else:
                         # Try scheduling the job
-                        if self._can_schedule(
-                                next_event.time, waiting_job_cpu_alloc, waiting_job_runtime, waiting_job_id
-                        ):
+                        if self._can_schedule(waiting_job_cpu_alloc):
                             self.wait_queue.popleft()
-                            self.mark_as_fixed_resource_job(waiting_job, time_waited_sec)
+                            self._schedule(next_event.time, waiting_job_cpu_alloc, waiting_job_runtime, waiting_job_id)
+                            self.set_waiting_time(waiting_job, time_waited_sec)
+                            waited_in_queue += 1
                         else:
                             break
             next_event = self._get_next_event()
 
-        self.calc_simulator_attributes()
+        self.calc_waiting_time_and_reserved_cost()
+        self.on_demand_cost = total_on_demand_cost
         self.print_results()
-
-    def calc_simulator_attributes(self):
-        self.input_trace["schedule"] = (
-            self.input_trace["job_id"].map(self.job_schedule_map)
-        )
-        self.input_trace["wait_time_sec"] = (
-            self.input_trace["job_id"].map(self.wait_time_map)
-        )
-        self.mean_waiting_time, self.on_demand_cost, self.reserved_cost = self.input_trace.wait_time_sec.mean(), \
-                                                                          self._compute_ondemand_cost(), self._compute_fixed_cost()
-
-    def mark_as_on_demand_resource_job(self, cur_job, waiting_time):
-        self.job_schedule_map[cur_job.job_id] = "D"
-        self.wait_time_map[cur_job.job_id] = waiting_time
-
-    def mark_as_fixed_resource_job(self, cur_job, waiting_time):
-        self.job_schedule_map[cur_job.job_id] = "R"
-        self.wait_time_map[cur_job.job_id] = waiting_time
-
-    """
-        Run the simulator (event based) with waiting queue. Applicable to LJW and compound policy.
-        """
+        print("Run immediately: " + str(run_immediately))
+        print("Run spec_exec correct: " + str(run_spec_exec_correct))
+        print("Run spec_exec incorrect: " + str(run_spec_exec_incorrect))
+        print("Waited in queue: " + str(waited_in_queue))
+        print("Waited max time: " + str(waited_max_time))
 
     def run_Ambati(
```

Comment:
Is it okay to use name Ambati on public repository? Should we name it as Original paper algorithm or something generic name referring previous paper?

Reply:
good call - i just renamed it to "original approach" accordingly. thanks!